src/Amazon.Common.DotNetCli.Tools/Utilities.cs

```diff
@@ -205,72 +206,125 @@ public static string DeterminePublishLocation(string workingDirectory, string pr
+                else
                 {
-                    msbuildProcessFailed = true;
+                    Console.WriteLine($"MSBuild process exited with code {process.ExitCode}. Error: {error}");
+                    // Fallback to XML parsing
+                    properties = LookupProjectPropertiesFromXml(projectFile, propertyNames);
                 }
             }
-            catch (Exception)
+            catch (Exception ex)
             {
-                // swallow any exceptions related to `dotnet msbuild`
-                msbuildProcessFailed = true;
+                Console.WriteLine($"Error executing MSBuild or parsing output: {ex.Message}");
+                // Fallback to XML parsing
+                properties = LookupProjectPropertiesFromXml(projectFile, propertyNames);
             }
 
-            if (msbuildProcessFailed)
+            return properties;
+        }
+
+        private static Dictionary<string, string> LookupProjectPropertiesFromXml(string projectFile, string[] propertyNames)
+        {
+            var properties = new Dictionary<string, string>();
+            try
             {
-                var projectFile = FindProjectFileInDirectory(projectLocation);
                 var xdoc = XDocument.Load(projectFile);
-                var element = xdoc.XPathSelectElement("//PropertyGroup/OutputType");
-                output = element?.Value;
+                foreach (var propertyName in propertyNames)
+                {
+                    var element = xdoc.XPathSelectElement($"//PropertyGroup/{propertyName}");
+                    if (element != null && !string.IsNullOrWhiteSpace(element.Value))
+                    {
+                        properties[propertyName] = element.Value;
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Error parsing project file XML: {ex.Message}");
             }
+            return properties;
+        }                                   
 
-            return 
-                string.IsNullOrEmpty(output) ? 
-                    null : 
-                    output.Trim();
+        /// <summary>
+        /// Looks up the target framework from a project file.
+        /// </summary>
+        /// <param name="projectLocation">The location of the project file.</param>
+        /// <returns>The target framework of the project.</returns>
+        public static string LookupTargetFrameworkFromProjectFile(string projectLocation)
+        {
+            var properties = LookupProjectProperties(projectLocation, "TargetFramework", "TargetFrameworks");
+            if (properties.TryGetValue("TargetFramework", out var targetFramework) && !string.IsNullOrEmpty(targetFramework))
+            {
+                return targetFramework;
+            }
+            if (properties.TryGetValue("TargetFrameworks", out var targetFrameworks) && !string.IsNullOrEmpty(targetFrameworks))
```

Comment:
The current system would throw an error if `TargetFrameworks` is used. I can see adding support for `TargetFrameworks` but only if resolves to a single value. We shouldn't just pick a framework which could be wrong but not discovered till after deployment. I'm sure if `TargetFrameworks` is used then it is rarely a single value but I have seen it sometimes after refactoring efforts. Say you drop a target but don't get around to changing back to `TargetFramework`.

So short response, if there is more then one framework in `TargetFrameworks` return `null` to stay consistent with previous behavior. In interactive mode that will prompt the user to choose the framework.

Reply:
sounds good i will update to that logic